wikidatarefisland/pipes/scraper.py

```diff
@@ -80,11 +80,13 @@ def _get_data_per_url(self, url, data, resourceUrls, mapping, item):
                         statement,
                         item['itemId'],
                         resourceUrls[url],
-                        datum[schema_property])
+                        datum[schema_property],
+                        url)
                     final_data.append(formatted)
         return final_data
 
-    def _format_result(self, statement, item_id, metadata, extracted_data):
+    def _format_result(self, statement, item_id, metadata, extracted_data, url):
```

Comment:
This is definitely needed for the contract; I'm wondering if we should move it outside of the referenceMetadata blob and into the parent "reference" object?
i.e. change the contract?

I think we are supposed to ensure that the URL isn't in the reference we add to Wikidata right? (see: https://phabricator.wikimedia.org/T251262#6096951)

Otherwise making change will also need us to update the populate script to filter this bit out again?

Reply:
The contract specifies that it should be outside of reference metadata (but I think it's not very clear): https://github.com/wmde/reference-island#pipe-2-scrape-given-urls-for-potential-matches-between--unreferenced-statements-and-structured-data

Reply:
The reference blob has it inside the blob: https://github.com/wmde/reference-island#referenceblob-format
*[referenceUrl]: [String],

Lydia said that url is not needed for the game API, that's why I had to make a PR to skip it.

Reply:
There is code in the api.php to skip that that I overlooked. see: wikidata_game/api.php:49

I'll merge this and add a ticket to update the docs. https://phabricator.wikimedia.org/T252640